array.go

```diff
@@ -7,6 +7,73 @@ import (
 )
 
 func (a *Asserter) checkArray(path string, act, exp []interface{}) {
+	a.tt.Helper()
+	if len(exp) > 0 && exp[0] == "<<UNORDERED>>" {
+		a.checkArrayUnordered(path, act, exp[1:])
+	} else {
+		a.checkArrayOrdered(path, act, exp)
+	}
+}
+
+func (a *Asserter) checkArrayUnordered(path string, act, exp []interface{}) {
+	a.tt.Helper()
+	if len(act) != len(exp) {
+		a.tt.Errorf("length of arrays at '%s' were different. Expected array to be of length %d, but contained %d element(s)", path, len(exp), len(act))
+		serializedAct, serializedExp := serialize(act), serialize(exp)
+		if len(serializedAct+serializedExp) < 50 {
+			a.tt.Errorf("actual JSON at '%s' was: %+v, but expected JSON was: %+v, potentially in a different order", path, serializedAct, serializedExp)
+		} else {
+			a.tt.Errorf("actual JSON at '%s' was:\n%+v\nbut expected JSON was:\n%+v,\npotentially in a different order", path, serializedAct, serializedExp)
+		}
+		return
+	}
+
+	for i, actEl := range act {
+		found := false
+		for _, expEl := range exp {
+			if a.deepEqual(actEl, expEl) {
+				found = true
+			}
+		}
+		if !found {
+			serializedEl := serialize(actEl)
+			if len(serializedEl) < 50 {
+				a.tt.Errorf("actual JSON at '%s[%d]' contained an unexpected element: %s", path, i, serializedEl)
+			} else {
+				a.tt.Errorf("actual JSON at '%s[%d]' contained an unexpected element:\n%s", path, i, serializedEl)
+			}
+		}
+	}
+
+	for i, expEl := range exp {
+		found := false
+		for _, actEl := range act {
+			if a.deepEqual(expEl, actEl) {
+				found = true
+			}
```

Comment:
Replace this with `found = found || a.deepEqual(expEl, actEl)` to avoid excessive (and expensive) calls to `a.deepEqual`.